hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapProxyImpl.java

```diff
@@ -92,6 +101,31 @@ public void initialize() {
         }
     }
 
+    @Override
+    public CompletionStage<Void> putAllAsync(@Nonnull Map<? extends K, Collection<? extends V>> m) {
+        checkTrueUnsupportedOperation(isClusterVersionGreaterOrEqual(Versions.V4_1), MINIMUM_VERSION_ERROR_4_1);
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        Map<Data, Data> dataMap = new HashMap<>();
+
+        for (Map.Entry e : m.entrySet()) {
+            dataMap.put(toData(e.getKey()), toData(e.getValue()));
```

Comment:
I just realized there is an issue with the serialization of the `Collection<? extends V> value` here:
- `Collection<? extends V>` is serialized to `Data`
- It will then be passed on as value to `MapEntries` field of `PutAllOperation`
- In [`PutAllOperation`](https://github.com/hazelcast/hazelcast/pull/16698/files#diff-a4c616539a59466e148996ae5c3b77e2R62) it is deserialized. The `Data` will be deserialized to `Collection<? extends V>`. The problem with this is that it requires that the collection's element class `Class<? extends V>` must be present on the server, even when the `MultiMap` stores items in `binary` format.

The solution is to first serialize each element, so the `Collection<? extends V>` is turned to a `Collection<Data>`, then perform the serialization of `Collection<Data>` to a single `Data` to be put in your `dataMap`:
```
for (Map.Entry e : m.entrySet()) {
  Collection<Data> dataCollection = asDataCollection(e.getValue());
  dataMap.put(toData(e.getKey()), toData(dataCollection));
}
...
Collection<Data> asDataCollection(Collection<? extends V> values) {
  // iterate values and serialize each
}
```